src/float.js

```diff
@@ -884,11 +884,13 @@ Sk.builtin.float_.prototype.str$ = function (base, sign) {
 
     if (base === undefined || base === 10) {
         if (Sk.__future__.python3) {
-            tmp = work.toPrecision(16);
+            // 17 would get another number string
+            var tmpCheckString = work.toString();
+            var tmpCheckPrecision = work.toPrecision(17);
```

Comment:
why can't we use `16` here?

Reply:
```javascript
var s = 0.1 + 0.2
s.toPrecision(17)
>>> "0.30000000000000004"
s.toPrecision(16)
>>> "0.3000000000000000"
```
If use 16, the tmp finally is 0.3 not 0.30000000000000004

Reply:
but when the number length more than 17:
```javascript
var s = 0.30000000000000000000000000004
s.toPrecision(17)
>>> "0.29999999999999999"
// except： 0.3
s.toString()
>>> 0.3
// this is right, but not all s.toString() is right, like str(1e1) -> 10.0
// but 1e1.toString() is 10 
// so I compare s.toString() with s.toPrecision(17) to guaranteed decimal.
```

Reply:
wow 😄